openerp/addons/base/ir/ir_attachment.py

```diff
@@ -431,6 +433,9 @@ def write(self, cr, uid, ids, vals, context=None):
         # remove computed field depending of datas
         for field in ['file_size', 'checksum']:
             vals.pop(field, False)
+        # if mimetype not given, compute it !
+        if 'mimetype' not in vals:
+            vals['mimetype'] = self._compute_mimetype(vals)
```

Comment:
Is this required ? This lines means the mimetype will be recompute every time a field will be modified on the attachment. The field should recompute when the data changed, which is a rare case (i think). Do you have a example ? Thanks